### PR TITLE
Hd tushar

### PR DESCRIPTION
**Pull request by gem-tusharmahajan**

### CODE REVIEW

File: src/test/java/com/qa/helpdesk/commonutils/Utils.java [hunk 1]
```diff
@@ -1,10 +1,11 @@
-package com.qa.helpdesk.commonutils;
+package helpdesk.commonutils;
 
 import com.gemini.generic.reporting.GemTestReporter;
 import com.gemini.generic.reporting.STATUS;
 import com.gemini.generic.ui.utils.DriverAction;
 import com.qa.helpdesk.locators.*;
-import com.qa.helpdesk.stepdefination.StepDefinition;
+import helpdesk.locators.*;
+import helpdesk.stepdefination.StepDefinition;
```
Comment (gem-rishabhrastogi) [inline src/test/java/com/qa/helpdesk/commonutils/Utils.java:8]: Use correct spelling for stepdefinition package

File: src/test/java/com/qa/helpdesk/locators/SearchInTickets.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.qa.helpdesk.locators;
+package helpdesk.locators;
```
Comment (gem-rishabhrastogi) [inline src/test/java/com/qa/helpdesk/locators/SearchInTickets.java:1]: -> Refrain from using hard-coded indexing while creating xpaths

File: src/test/java/com/qa/helpdesk/stepdefination/Hook.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package com.qa.helpdesk.stepdefination;
+package helpdesk.stepdefination;
```
Comment (gem-rishabhrastogi) [inline src/test/java/com/qa/helpdesk/stepdefination/Hook.java:1]: Please follow proper naming conventions [Start() -> start()]

**Review by gem-rishabhrastogi — CHANGES_REQUESTED** [review]
Added some suggestions. Please work on those.

File: src/test/java/com/qa/helpdesk/commonutils/Utils.java [hunk 4]
```diff
@@ -345,11 +344,11 @@ public void clickBtn(String btnName)
         }
         if (elementPresent)
         {
-        GemTestReporter.addTestStep("Verify button is clicked",btnName+ "button is present and clicked",STATUS.PASS);
+        GemTestReporter.addTestStep("Verify button is clicked",btnName+ " button is present and clicked",STATUS.PASS,DriverAction.takeSnapShot());
         }
    else
     {
-    GemTestReporter.addTestStep("Verify button is clicked",btnName+ "button is not present and clicked",STATUS.FAIL);
+    GemTestReporter.addTestStep("Verify button is clicked",btnName+ " button is not present and clicked",STATUS.FAIL,DriverAction.takeSnapShot());
```
Comment (gem-rishabhrastogi) [inline src/test/java/com/qa/helpdesk/commonutils/Utils.java:351]: Only 'btnName+" is not present" is enough.

File: src/test/java/com/qa/helpdesk/stepdefinition/StepDefinition.java [hunk 5]
```diff
@@ -0,0 +1,1905 @@
...
+        boolean filterTab = DriverAction.isExist(Filter.validateFilterTab);
+        if (filterTab) {
+            GemTestReporter.addTestStep("validate filter tab", "filter tab is validated", STATUS.PASS, DriverAction.takeSnapShot());
+        } else {
+            GemTestReporter.addTestStep("validate filter tab", "filter tab is not validated", STATUS.FAIL, DriverAction.takeSnapShot());
+        }
+    }
+
+
+    @And("^Select \"([^\"]*)\" and select \"([^\"]*)\" in filter tab$")
```
Comment (Gem-NeerajSingh) [inline src/test/java/com/qa/helpdesk/stepdefinition/StepDefinition.java:190]: {String}

**Review by Gem-NeerajSingh — CHANGES_REQUESTED** [review]
changes req

**Pull request closed without merging**